deepchem/models/torch_models/grover.py

```diff
@@ -92,3 +98,312 @@ def forward(self, graph_batch: List[torch.Tensor]):
+                self.readout(embeddings["atom_from_atom"], atom_scope))
+        if embeddings["atom_from_bond"] is not None:
+            preds_atom_from_bond = self.linear_atom_from_bond(
+                self.readout(embeddings["atom_from_bond"], atom_scope))
+
+        return {
+            "atom_from_atom": preds_atom_from_atom,
+            "atom_from_bond": preds_atom_from_bond,
+            "bond_from_atom": preds_bond_from_atom,
+            "bond_from_bond": preds_bond_from_bond
+        }
+
+
+class GroverPretrainer(ModularTorchModel):
+
+    def __init__(self,
+                 atom_vocab_size,
+                 bond_vocab_size,
+                 fg_size,
+                 hidden_size=128,
+                 dropout=0.2,
+                 activation='relu',
+                 num_mt_block=1,
+                 num_attn_head=1,
+                 bias=True):
+        self.model = self.build_model(hidden_size, dropout, activation,
+                                      num_mt_block, num_attn_head, bias)
+        self.components = self.build_components(hidden_size, atom_vocab_size,
+                                                bond_vocab_size, fg_size)
+
+    def build_model(self, hidden_size, dropout, activation, num_mt_block,
+                    num_attn_head, bias):
+        return GroverEmbedding(embedding_output_type='both',
+                               hidden_size=hidden_size,
+                               dropout=dropout,
+                               activation=activation,
+                               num_mt_block=num_mt_block,
+                               num_attn_head=num_attn_head,
+                               bias=bias)
+
+    def build_components(self, hidden_size, atom_vocab_size, bond_vocab_size,
+                         fg_size):
+        components = {}
+        # layer for predicting atom vocabulary using atom emeddings
+        components['atom_vocab_task_atom'] = self._atom_vocab_prediction_model(
+            atom_vocab_size, hidden_size)
+        # layer for predicting atom vocabulary using bond emeddings
+        components['atom_vocab_task_bond'] = self._atom_vocab_prediction_model(
+            atom_vocab_size, hidden_size)
+        # layer for predicting bond vocabulary using atom embeddings
+        components['bond_vocab_task_atom'] = _BondVocabPredictionLayer(
+            bond_vocab_size, hidden_size)
+        # layer for predicting bond vocabulary using bond embeddings
+        components['bond_vocab_task_bond'] = _BondVocabPredictionLayer(
+            bond_vocab_size, hidden_size)
+        # layer for predicting functional group
+        components['fg_task_all'] = _FunctionalGroupPrediction(fg_size)
+        return components
+
+    def _atom_vocab_prediction_layer(self, vocab_size, hidden_size):
+        return nn.Sequential(nn.Linear(hidden_size, vocab_size),
+                             nn.LogSoftmax(dim=1))
+
+    def loss_func(self, inputs, labels, weights):
```

Comment:
Can you make the loss a deepchem loss? @tonydavis629 and I have discussed that we should set a new convention that losses should be added with other deepchem losses forr `ModularTorchModel`